polar-core/src/partial/simplify.rs

```diff
@@ -0,0 +1,217 @@
+use std::collections::HashSet;
```

Comment:
Not going to do a thorough review of this in the interest of time and because it's being taken as a follow-up.